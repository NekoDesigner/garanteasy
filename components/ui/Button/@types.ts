export interface IButtonProps {
  label: string;
  onPress?: () => void;
  disabled?: boolean;
  style?: object;
  testID?: string;
  size?: "s" | "xs";
  variant?: TButtonVariant;
  icon?: React.ReactNode;
  showIcon?: boolean;
  iconPosition?: "left" | "right";
}

export type TButtonVariant =
  | "primary"
  | "secondary"
  | "danger"
  | "outline-primary"
  | "outline-secondary"
  | "outline-danger";
