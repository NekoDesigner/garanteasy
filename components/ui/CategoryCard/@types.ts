import { PropsWithChildren } from 'react';
export interface ICardProps extends PropsWithChildren {
  style?: object;
  testID?: string;
  onPress?: () => void;
}