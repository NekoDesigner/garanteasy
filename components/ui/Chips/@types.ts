export type TChipsIcon =
  | "hoursehold-electricals"
  | "small-electricals"
  | "diy"
  | "garden"
  | "fashion"
  | "other";

export interface IChipsProps {
  category: TChipsIcon | React.ReactNode;
  text: string;
  size?: "s" | "xs";
  showIcon?: boolean;
  onPress?: () => void;
}
