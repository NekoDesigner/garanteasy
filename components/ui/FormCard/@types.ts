/**
 * IFormCardProps - Generated by Garanteasier CLI
 */

import { PropsWithChildren } from "react";

export interface IFormCardProps extends PropsWithChildren {
  style?: object;
  testID?: string;
}