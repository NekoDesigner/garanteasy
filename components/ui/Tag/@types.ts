export interface ITagProps {
  label: string;
  size?: 'small' | 'normal';
  style?: object;
  textStyle?: object;
  icon?: React.ReactNode;
  showIcon?: boolean;
  iconPosition?: 'left' | 'right';
  testID?: string;
}
