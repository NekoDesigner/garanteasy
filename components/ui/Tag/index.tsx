import React from "react";
import { View, Text } from "react-native";
import { ITagProps } from "./@types";
import TagStyles from "./styles";

const Tag: React.FC<ITagProps> = ({ style = {}, testID = 'tag', size = 'normal', textStyle = {}, label, showIcon = true, icon, iconPosition = 'left' }) => {

  const styles = React.useMemo(() => {
    return TagStyles({ size });
  }, [size]);

  const renderIcon = React.useMemo(() => {
    if (!showIcon || !icon || !React.isValidElement(icon)) return null;

    return React.cloneElement(
      icon as React.ReactElement<any>,
      {
        style: [
          styles.icon,
          (icon.props && typeof icon.props === 'object' && icon.props !== null && 'style' in icon.props
            ? icon.props.style
            : undefined),
        ],
        testID: `${testID}-icon`,
        color: styles.icon.color, // Ensure the icon color matches the button style
      }
    );
  }, [showIcon, icon, styles.icon, testID]);

  return (
    <View style={[styles.container, style]} testID={testID}>
      {showIcon && iconPosition === "left" && renderIcon}
      <Text style={[styles.text, textStyle]} testID={`${testID}-text`}>{label}</Text>
      {showIcon && iconPosition === "right" && renderIcon}
    </View>
  );
};

export default Tag;
