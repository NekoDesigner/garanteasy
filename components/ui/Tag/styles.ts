import { StyleSheet } from 'react-native';
import { COLORS, SIZES } from '../../../constants';
import { ITagProps } from './@types';

function getFontSizeStyles(size: 'small' | 'normal') {
  switch (size) {
    case 'normal':
      return SIZES.font.xs;
    case 'small':
      return SIZES.font.xxs;
    default:
      return SIZES.font.xs;
  }
}

const styles = ({ size = 'normal' }: Omit<ITagProps, 'label'>) => (StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: SIZES.padding.xxs,
    paddingHorizontal: SIZES.padding.xs,
    gap: SIZES.padding.xxs,
    borderRadius: SIZES.padding.xxs,
    backgroundColor: COLORS.light,
  },
  text: {
    fontSize: getFontSizeStyles(size),
    color: COLORS.blueDarker,
    fontFamily: SIZES.font.familly.default,
    fontWeight: SIZES.font.weight.semiBold,
  },
  icon: {
    width: SIZES.icon.xs,
    height: SIZES.icon.xs,
    color: COLORS.blueDarker
  }
}));

export default styles;
