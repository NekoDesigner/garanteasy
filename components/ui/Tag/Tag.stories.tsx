import React from "react";
import { View } from "react-native";
import ScanIcon from "../Icons/ScanIcon";
import Tag from "./index";

export default {
  title: 'Tag',
  component: Tag
};

export const Default = () => (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', padding: 16, gap: 24 }}>
      <Tag label='Expire dans 40j'icon={<ScanIcon />} />
    </View>
);