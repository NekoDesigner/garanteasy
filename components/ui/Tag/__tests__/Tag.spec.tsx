import { render } from "@testing-library/react-native";
import React from "react";
import ScanIcon from "../../Icons/ScanIcon";
import { ITagProps } from "../@types";
import Tag from "../index";

describe("Tag Components", () => {

  const DEFAULT_TAG_PROPS: ITagProps = {
    label: "Default Tag",
    showIcon: true,
    icon: <ScanIcon />,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('render tag correctly with default props', () => {
    const { getByTestId } = render(<Tag {...DEFAULT_TAG_PROPS} />);
    const tagElement = getByTestId('tag');
    const textElement = getByTestId('tag-text');
    const iconElement = getByTestId('tag-icon');
    expect(tagElement).toBeTruthy();
    expect(textElement).toBeTruthy();
    expect(iconElement).toBeTruthy();
    expect(textElement.props.children).toBe(DEFAULT_TAG_PROPS.label);
  });

  it('renders tag without icon when showIcon is false', () => {
    const { getByTestId, queryByTestId } = render(
      <Tag {...DEFAULT_TAG_PROPS} showIcon={false} />
    );
    const tagElement = getByTestId('tag');
    const textElement = getByTestId('tag-text');
    const iconElement = queryByTestId('tag-icon');
    expect(tagElement).toBeTruthy();
    expect(textElement).toBeTruthy();
    expect(iconElement).toBeNull(); // Icon should not be rendered
  });

  it('renders tag with custom style', () => {
    const customStyle = { backgroundColor: 'blue', padding: 10 };
    const { getByTestId } = render(
      <Tag {...DEFAULT_TAG_PROPS} style={customStyle} />
    );
    const tagElement = getByTestId('tag');
    expect(tagElement.props.style).toContainEqual(customStyle);
  });
  it('renders tag with custom text style', () => {
    const customTextStyle = { color: 'red', fontSize: 20 };
    const { getByTestId } = render(
      <Tag {...DEFAULT_TAG_PROPS} textStyle={customTextStyle} />
    );
    const textElement = getByTestId('tag-text');
    expect(textElement.props.style).toContainEqual(customTextStyle);
  });

    test('renders correctly', () => {
      const tree = render(<Tag label='Content' />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});