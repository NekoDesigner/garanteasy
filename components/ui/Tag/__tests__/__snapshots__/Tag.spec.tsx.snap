// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Components renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 4,
        "display": "flex",
        "flexDirection": "row",
        "gap": 4,
        "justifyContent": "space-between",
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
      {},
    ]
  }
  testID="tag"
>
  <Text
    style={
      [
        {
          "color": "#050426",
          "fontFamily": "Ubuntu",
          "fontSize": 12,
          "fontWeight": "700",
        },
        {},
      ]
    }
    testID="tag-text"
  >
    Content
  </Text>
</View>
`;
